From 4bcf8ff4618e3d7e0ea535687ffe0e3522fa3b69 Mon Sep 17 00:00:00 2001
From: Christopher Champagne <cjchampagne@habana.ai>
Date: Thu, 22 May 2025 14:43:11 +0300
Subject: [PATCH] [SW-179783] g3: pldm: update sensor read rates

Update sensor polling rates to high (2.5 seconds) or very high (1 second)
based on spreadsheet changes.

Change-Id: I718e9439778fccc670ac48dffc722dc2f86f54ab
Signed-off-by: Christopher Champagne <cjchampagne@habana.ai>
---

diff --git a/arch/gaudi3/common/ras/pldm/sensord_db.c b/arch/gaudi3/common/ras/pldm/sensord_db.c
index ab5fa09..8a0cc4b 100644
--- a/arch/gaudi3/common/ras/pldm/sensord_db.c
+++ b/arch/gaudi3/common/ras/pldm/sensord_db.c
@@ -12,6 +12,7 @@
 	{
 		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_CURRENT_VRM_MAX_TEMP],
 		.api.poll = sensord_poll_current_vrm_max_temp,
+		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
 	},
 	{
 		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_OAM_LAST_POWER_DOWN_CAUSE_MAIN],
@@ -65,6 +66,7 @@
 	{
 		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_CURRENT_SOC_54V_POWER],
 		.api.poll = sensord_poll_current_soc_54v_power,
+		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
 	},
 	{
 		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_COMPOSITE_TEMPERATURE],
@@ -85,6 +87,7 @@
 	{
 		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_PCIE_UNCORRECTABLE_ERROR_STATE_FATAL],
 		.api.get = sensord_get_pcie_uncorrectable_error_state_fatal,
+		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
 	},
 	{
 		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_CHECKSUM_ERR_STORAGE_ERR_TBLS_STATE],
@@ -101,12 +104,12 @@
 	{
 		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_DEVICE_UTILIZATION],
 		.api.poll = sensord_poll_device_utilization,
-		.api.polling_rate_ms = 1000,
+		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_VERY_HIGH,
 	},
 	{
 		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_HEALTH_STATE],
 		.api.poll = sensord_poll_health_state,
-		.api.polling_rate_ms = 1000,
+		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_VERY_HIGH,
 	},
 	{
 		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_FW_STATE],
@@ -115,6 +118,7 @@
 	{
 		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_POWER_AND_PERFORMANCE_STATE],
 		.api.get = sensord_get_power_and_performance_state,
+		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
 	},
 	{
 		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_ERROR_STATE_1],
